internal/rukpak/source/containers_image_test.go

```diff
@@ -0,0 +1,400 @@
+package source_test
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/BurntSushi/toml"
+	"github.com/containers/image/v5/docker/reference"
+	"github.com/containers/image/v5/pkg/sysregistriesv2"
+	"github.com/containers/image/v5/types"
+	"github.com/google/go-containerregistry/pkg/crane"
+	"github.com/olareg/olareg"
+	"github.com/olareg/olareg/config"
+	"github.com/opencontainers/go-digest"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/operator-framework/operator-controller/internal/rukpak/source"
+)
+
+const (
+	testFileName     string = "test-file"
+	testFileContents string = "test-content"
+)
+
+func TestUnpackValidInsecure(t *testing.T) {
+	imageTagRef, _, cleanup := setupRegistry(t)
+	defer cleanup()
+
+	unpacker := &source.ContainersImageRegistry{
+		BaseCachePath: t.TempDir(),
+		SourceContext: buildPullContext(t, imageTagRef),
+	}
+	bundleSource := &source.BundleSource{
+		Name: "test-bundle",
+		Type: source.SourceTypeImage,
+		Image: &source.ImageSource{
+			Ref: imageTagRef.String(),
+		},
+	}
+
+	oldBundlePath := filepath.Join(unpacker.BaseCachePath, bundleSource.Name, "old")
+	err := os.MkdirAll(oldBundlePath, 0755)
+	require.NoError(t, err)
+
+	// Attempt to pull and unpack the image
+	result, err := unpacker.Unpack(context.Background(), bundleSource)
+	// Check Result
```

Comment:
I think all the `// Check Result` comments might be a little redundant...

Reply:
Copy/pasta from the original test suite. I can remove.